base/sharedarray.jl

```diff
@@ -148,14 +148,17 @@ function SharedArray{T,N}(filename::AbstractString, ::Type{T}, dims::NTuple{N,In
 
     # If not supplied, determine the appropriate mode
     have_file = onlocalhost ? isfile(filename) : remotecall_fetch(isfile, pids[1], filename)
-    if mode == nothing
+    if mode === nothing
         mode = have_file ? "r+" : "w+"
     end
     workermode = mode == "w+" ? "r+" : mode  # workers don't truncate!
 
     # Ensure the file will be readable
     mode in ("r", "r+", "w+", "a+") || throw(ArgumentError("mode must be readable, but $mode is not"))
-    init==false || mode in ("r+", "w+", "a+") || throw(ArgumentError("cannot initialize unwritable array (mode = $mode)"))
+    if init !== false
+        typeassert(init, Function)
```

Comment:
good call. I added this so that `init` isn't silently ignored later (https://github.com/JuliaLang/julia/pull/16764/files/e6e1312a051d9b5511bf5b4f9ea0ae97528bb5ed#diff-6ba74fe8f18fa73bec8dfbfd1801a05cR209), but that test is wrong too